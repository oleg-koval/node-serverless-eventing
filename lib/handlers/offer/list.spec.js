jest.mock('../../models/loan')

const { handler } = require('./list')
const LoanModel = require('../../models/loan')

beforeEach(() => {
    jest.resetAllMocks()
})

describe('list.handler', async () => {
    it('Should list loans', async () => {
        LoanModel.Loan.findAll.mockImplementation(cb => {
            cb(null, {
                Items: [
                    {
                        createdAt: '2019-06-12T00:11:44.898Z',
                        amount: 333,
                        id: '052258d8-670b-40b7-8658-82a9855b412a',
                        status: 'offered',
                    },
                    {
                        createdAt: '2019-06-12T00:11:02.473Z',
                        amount: 333,
                        id: 'f3f2beb4-c9f7-4329-8f71-faea34073d14',
                        status: 'offered',
                    },
                ],
            })
        })

        const result = await handler()
        expect(result).toMatchSnapshot()
    })
    it('Return error', async () => {
        LoanModel.Loan.findAll.mockImplementation(cb => {
            cb(new Error('problem with dynamo'))
        })

        const result = await handler()
        expect(result).toMatchSnapshot()
    })
    it('Return empty result', async () => {
        LoanModel.Loan.findAll.mockImplementation(cb => {
            cb(null, { Items: [] })
        })

        const result = await handler()
        expect(result).toMatchSnapshot()
    })
})
