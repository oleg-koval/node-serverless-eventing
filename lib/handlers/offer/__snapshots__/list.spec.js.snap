// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list.handler Return empty result 1`] = `
Object {
  "body": Array [],
  "headers": Object {},
  "statusCode": 200,
}
`;

exports[`list.handler Return error 1`] = `
Object {
  "body": [Error: problem with dynamo],
  "headers": Object {},
  "statusCode": 400,
}
`;

exports[`list.handler Should list loans 1`] = `
Object {
  "body": Array [
    Object {
      "amount": 333,
      "createdAt": "2019-06-12T00:11:44.898Z",
      "id": "052258d8-670b-40b7-8658-82a9855b412a",
      "status": "offered",
    },
    Object {
      "amount": 333,
      "createdAt": "2019-06-12T00:11:02.473Z",
      "id": "f3f2beb4-c9f7-4329-8f71-faea34073d14",
      "status": "offered",
    },
  ],
  "headers": Object {},
  "statusCode": 200,
}
`;
